src/python/ensembl/io/genomio/manifest/generate.py

```diff
@@ -73,26 +75,51 @@ def get_files_checksums(self):
                 continue
 
             for name, standard_name in self.names.items():
-                if subfile.stem.endswith(name):
+                # Either the last element of the stem or the suffix is a known name
+                if subfile.stem.endswith(name) or subfile.suffix == f".{name}":
+                    logging.debug(f"Matched to {name} ({standard_name}) = {subfile}")
                     used_file = True
                     md5 = self._get_md5sum(subfile)
                     file_obj = {"file": subfile.name, "md5sum": md5}
-                    if standard_name in manifest_files:
-                        if isinstance(manifest_files[standard_name], list):
-                            manifest_files[standard_name].append(file_obj)
-                        else:
-                            # Convert to a list
-                            manifest_files[standard_name] = [manifest_files[standard_name], file_obj]
 
+                    # Multiple files stored, each with a name
+                    if standard_name in self.multi_files:
+                        manifest_files.setdefault(standard_name, {})
+                        obj_name = self._prepare_object_name(subfile, name, manifest_files[standard_name])
+                        manifest_files[standard_name][obj_name] = file_obj
+
+                    # Single file
                     else:
                         manifest_files[standard_name] = file_obj
-                    break
 
             if not used_file:
                 logging.warning(f"File {subfile} was not included in the manifest")
 
         return manifest_files
 
+    def _prepare_object_name(self, subfile: Path, name: str, manifest_dict: dict) -> str:
+        # Prepare object name
+        obj_name = "file"
+        try:
+            # If we recognize the suffix, then the name is the part after the last "_"
+            if subfile.suffix == f".{name}":
+                obj_name = subfile.stem.split(sep="_")[-1]
+            # If we recognize the end of the name, then the name is the part before the last "_"
+            else:
+                obj_name = subfile.stem.split(sep="_")[-2]
+        except IndexError:
+            pass
+
+        # Add number if duplicate name
+        obj_name_base = obj_name
+        count = 1
+        while obj_name in manifest_dict:
+            obj_name = f"{obj_name_base}.{count}"
+            count += 1
+            if count >= 10:
```

Comment:
Any reason for 10 files being the threshold ?

Reply:
Just an arbitrary number in case there are a lot of files. If we encounter a database with a lot of those (right now only from AGP) then we can update the number.

Reply:
Shall we add this number as an argument for the function, with default value set to 10? This way it can be easily changed for specific cases/conditions if needed.